### PR TITLE
Jhassler/octocam

### PR DESCRIPTION
**Pull request by JoshuaHassler**
Refactored camera to take advantage of c++ stuff. It should now be easier to work with and integrate into OctopOS.

### CODE REVIEW

**Comment by JoshuaHassler** [comment]
needs to be tested on the adrv

**Comment by JoshuaHassler** [comment]
also we may want to port over some more of the comments

**Review by simius — COMMENTED** [review]
are we intentional declaring INCLUDE_CAMERA_H? As opposed to just CAMERA_H?

**Comment by simius** [comment]
> also we may want to port over some more of the comments

More comments from Sean's original code? I think most of his stuff was for his own debugging purposes, rather than useful for the code base.

**Review by simius — DISMISSED** [review]

**Comment by JoshuaHassler** [comment]
so, technically if we are following Google Code Style guidelines ( which was the original intent of the project ) your header guard should reference the path up the nearest .git folder. As a header guard, it doesn't particularly matter as long as it is consistent and there is no collision with other files.

tl;dr; I was following the style guidelines we technically have. [Some Old Docs](https://tree.taiga.io/project/darrieng-command-and-data-handling/wiki/home), [Our tool for linting](https://github.com/SPACE-HAUC/Powerlint)

**Comment by JoshuaHassler** [comment]
I'll add that the code style guidelines are only strictly enforced for [OctopOS](https://github.com/SPACE-HAUC/OctopOS)

File: src/main.cpp [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+#include <memory>
+#include <iostream>
+#include "camera.h"
+#include "mock.h"
+#include "ucam.h"
+
+typedef struct {
+    bool takePicture;
+    char imgLogcation[50];
+} CameraType;
```
Comment (kodycode) [inline src/main.cpp:12]: Is `CameraType` still needed? I don't see a reference to this anywhere.

**Review by sjc625 — APPROVED** [review]
Looks good, base camera functions wrapped in C++ with CMake added. Will test on hardware ASAP.

File: src/main.cpp (hunk 1 above)
Comment (simius) [reply src/main.cpp:12]: @JoshuaHassler @sjc625 ?

Comment (JoshuaHassler) [reply src/main.cpp:12]: It will probably be when I integrate it with octopos
